core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/HDTParser.java

```diff
@@ -0,0 +1,311 @@
+import org.eclipse.rdf4j.rio.RioSetting;
+
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionary;
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionarySection;
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionarySectionFactory;
+import org.eclipse.rdf4j.rio.hdt.part.HDTGlobal;
+import org.eclipse.rdf4j.rio.hdt.part.HDTHeader;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriples;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriplesSection;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriplesSectionFactory;
+
+import org.eclipse.rdf4j.rio.helpers.AbstractRDFParser;
+
+/**
+ * RDF parser for HDT v1.0 files. This parser is not thread-safe, therefore its public methods are synchronized.
+ * 
+ * Unfortunately the draft specification is not entirely clear and probably slightly out of date, since the open source
+ * reference implementation HDT-It seems to implement a slightly different version. This parser tries to be compatible
+ * with HDT-It 1.0.
+ * 
+ * The most important parts are the Dictionaries containing the actual values (S, P, O part of a triple), and the
+ * Triples containing the numeric references to construct the triples.
+ * 
+ * Since objects in one triple are often subjects in another triple, these "shared" parts are stored in a shared
+ * Dictionary, which may significantly reduce the file size.
+ * 
+ * File structure:
+ * 
+ * <pre>
+ * +---------------------+
+ * | Global              |
+ * | Header              |
+ * | Dictionary (Shared) |
+ * | Dictionary (S)      |
+ * | Dictionary (P)      |
+ * | Dictionary (O)      |    
+ * | Triples             |
+ * +---------------------+
+ * </pre>
+ * 
+ * @author Bart Hanssens
+ * 
+ * @see <a href="http://www.rdfhdt.org/hdt-binary-format/">HDT draft (2015)</a>
+ * @see <a href="https://www.w3.org/Submission/2011/03/">W3C Member Submission (2011)</a>
+ */
+public class HDTParser extends AbstractRDFParser {
+	// buffer for input stream
+	private final static int BUFLEN = 16384;
+
+	private Map<String, String> globalProps;
+	private Map<String, String> headerProps;
+	private byte[] headerData;
+
+	/**
+	 * Creates a new HDTParser that will use a {@link SimpleValueFactory} to create RDF model objects.
+	 */
+	public HDTParser() {
+		super();
+	}
+
+	/**
+	 * Creates a new HDTParser that will use the supplied ValueFactory to create RDF model objects.
+	 *
+	 * @param valueFactory A ValueFactory.
+	 */
+	public HDTParser(ValueFactory valueFactory) {
+		super(valueFactory);
+	}
+
+	@Override
+	public RDFFormat getRDFFormat() {
+		return RDFFormat.HDT;
+	}
+
+	@Override
+	public Collection<RioSetting<?>> getSupportedSettings() {
+		Set<RioSetting<?>> result = new HashSet<>();
+		return result;
+	}
+
+	/**
+	 * Get global properties
+	 * 
+	 * @return key,value map
+	 */
+	public Map<String, String> getGlobalProperties() {
+		return globalProps;
+	}
+
+	/**
+	 * Get header properties
+	 * 
+	 * @return key,value map
+	 */
+	public Map<String, String> getHeaderProperties() {
+		return headerProps;
```

Comment:
This is the only reference, read or write, to this field.

Reply:
no use for it (yet), so I'll remove it